Library/Homebrew/dev-cmd/typecheck.rb

```diff
@@ -47,6 +52,21 @@ def typecheck
         return
       end
 
+      if args.prune_files_list?
+        YAML.parse(File.read(SORBET_FILES_YAML)).each do |line|
+          next unless line.scalar? && line.value.end_with?(".rb")
+
+          filepath = line.value
+
+          unless File.exist?(filepath)
+            odebug "Deleting #{filepath} from #{SORBET_FILES_YAML} as it doesn't exist."
+            `sed -i '' '/#{filepath.gsub!("/", "\\/")}/d' #{SORBET_FILES_YAML}`
```

Comment:
Would be nice to do this in plain Ruby; build a list of files to be removed, read the file line by line and don't output the lines of missing files and write it to disk.

Reply:
I removed the YAML parsing because it was getting in the way of writing the file back properly (I needed the raw data, not the parsed data), dealt with the Sorbet `true`, `false`, `strict`, `strong` keywords separately, then write every line of the file that _can stay_ to an array (that was easier to model in my brain), then write the file at the end.